plugin/reload/setup.go

```diff
@@ -73,9 +74,11 @@ func setup(c *caddy.Controller) error {
 	})
 
 	// re-register on finalShutDown as the instance most-likely will be changed
-	c.OnFinalShutdown(func() error {
-		r.quit <- true
-		return nil
+	shutOnce.Do(func() {
```

Comment:
Yes. we want only one QUIT signal to stop the onTimer loop of the last Hook.

Reply:
Anyway all these onTimer launched on each reload, even though would not have stopped as expected, are listening the SAME channel. One QUIT is enough.
Two QUIT will block the sender and then the application cannot shutdown properly.

Reply:
so now it *doesn't* re-register this, like the comment says?